src/Pages/Rooms/Rooms.tsx

```diff
@@ -0,0 +1,200 @@
+import { defaultImage } from '@/Assets/Images';
+import { TableHeader } from '@/Components';
+import { RoomsData } from '@/Redux/Features/Rooms/GetRoomsSlice';
+import { DataGrid, GridColDef, GridToolbar } from '@mui/x-data-grid';
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import './Rooms.module.scss';
+
+
+
+const Rooms = () => {
+
+  const dispatch = useDispatch();
```

Comment:
It is considered a suboptimal practice to directly utilize dispatcher functions within components. Drawing from insights shared in a talk by Mark Erkison, a recommended approach involves creating custom hooks. These hooks serve as an intermediary where dispatchers are integrated, and subsequently, the hooks are employed throughout the application. This modular approach aligns with best practices, enhancing code organization and maintainability.